microcosm/opaque.py

```diff
@@ -39,9 +42,17 @@ def __enter__(self):
             self.saved = deepcopy(opaque._store)
             opaque.update(self.func())
 
+            if opaque.tracer:
+                span = self.enter_context(opaque.tracer.start_span(opaque.name))
```

Comment:
Using `opaque.name` for the span name is my least favorite part. I didn't want to chase around for a nicer introspection invocation to find a file or function name that should be close to whereever this code is running, but I think such an invocation should exist.